FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -18,7 +18,44 @@ public final class RemoteFeedLoader: FeedLoader {
 		self.client = client
 	}
 
+	private static var OK_200: Int { return 200 }
+
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { result in
+			switch result {
+			case let .success((data, httpURLResponse)):
+
+				if httpURLResponse.statusCode != RemoteFeedLoader.OK_200 {
+					completion(.failure(Error.invalidData))
+					return
+				}
+
+				guard let value = try? JSONDecoder().decode(RemoteFeedItem.self, from: data) else {
+					completion(.failure(Error.invalidData))
+					return
+				}
```

Comment:
It's possible to combine both the `if` statement and the `guard` clause, having a single path for success and a single path for failure, instead of having two places that complete with `.invalidData.`
```suggestion
				guard httpURLResponse.statusCode == RemoteFeedLoader.OK_200,
				      let value = try? JSONDecoder().decode(RemoteFeedItem.self, from: data) else {
					return completion(.failure(Error.invalidData))
				}
```
Also, consider moving validation and/or mapping logic to helper functions, to make the `load(completion:)` function easier to read.